track_and_trace/asset_client/src/views/signup_form.js

```diff
@@ -23,84 +23,197 @@ const forms = require('../components/forms')
-  const user = _.assign(keys, _.pick(state, 'username', 'email'))
-  user.password = api.hashPassword(state.password)
-  const agent = payloads.createAgent(_.pick(state, 'name'))
+    AgentSignUp.username = ''
+    AgentSignUp.password = ''
+    AgentSignUp.passwordConfirm = ''
+    AgentSignUp.name = ''
+    AgentSignUp.organization = ''
+  },
 
-  transactions.submit(agent, true)
-    .then(() => api.post('users', user))
-    .then(res => api.setAuth(res.authorization))
-    .then(() => m.route.set('/'))
+  invalid: () => {
+    if (!AgentSignUp.username ||
+        AgentSignUp.password !== AgentSignUp.passwordConfirm ||
+        !AgentSignUp.name) {
+          return true
+    }
+
+    return false
+  }
 }
 
 /**
- * The Form for authorizing an existing user.
+ * Agent Sign Up Form
  */
-const SignupForm = {
-  view (vnode) {
-    const setter = forms.stateSetter(vnode.state)
-
-    return m('.signup-form', [
-      m('form', { onsubmit: userSubmitter(vnode.state) },
-      m('legend', 'Create Agent'),
-      forms.textInput(setter('name'), 'Name'),
-      forms.emailInput(setter('email'), 'Email'),
-      forms.textInput(setter('username'), 'Username'),
-      passwordCard(vnode.state),
-      m('.container.text-center',
-        'Or you can ',
-        m('a[href="/login"]',
-          { oncreate: m.route.link },
-          'login an existing Agent')),
-      m('.form-group',
-        m('.row.justify-content-end.align-items-end',
-          m('col-2',
-            m('button.btn.btn-primary',
-              'Create Agent')))))
-    ])
+const AgentSignupForm = {
+  oninit() {
+    AgentSignUp.clear()
+  },
+  view() {
+    return [
+      m('.signup-form'),
+        m('form', [
+            AgentSignUp.error ? m('p.text-danger', AgentSignUp.error) : null,
+            m('legend', 'Create Agent'),
+            inputField('username', 'Email', AgentSignUp.username, AgentSignUp.setUsername),
+            inputField('password', 'Password', AgentSignUp.password, AgentSignUp.setPassword, 'password'),
+            inputField('passwordConfirm', 'Confirm Password', AgentSignUp.passwordConfirm, AgentSignUp.setPasswordConfirm, 'password'),
+            inputField('name', 'Name', AgentSignUp.name, AgentSignUp.setName),
+            inputField('organization', 'Organization', AgentSignUp.organization, AgentSignUp.setOrganization),
+
+            m('.container.text-center',
+              m('a[href="/login"]',
+                { oncreate: m.route.link },
+                'login an existing Agent')),
+            m('.form-group',
+              m('.row.justify-content-end.align-items-end',
+                m('col-2',
+                  m('button.btn.btn-primary',
+                    {
+                      onclick: AgentSignUp.submit,
+                      disabled: AgentSignUp.submitting || AgentSignUp.invalid(),
+                    }, 'Create Agent'))))
+        ])
+    ]
   }
 }
 
-module.exports = SignupForm
+// const passwordCard = state => {
```

Comment:
Please do not include commented out code in your commits.

Reply:
Commented block removed